pyvista/__init__.py

```diff
@@ -60,6 +63,17 @@
 MAX_N_COLOR_BARS = 10
 
 
+# Import all modules for type checkers and linters
+if TYPE_CHECKING:  # pragma: no cover
+    import demos
+    import ext
+    from plotting import *
```

Comment:
Does this mean that anyone using an IDE with auto complete will now need to import GL libraries?

Reply:
No, the type checkers do not actually execute any code. So this will not force any imports to actually be imported.

If the libraries are not available, then linters may flag it as an error similar to this one in PyVista's `tools.opacity_transfer_function`, where `scipy` is imported by is not installed:
![image](https://github.com/pyvista/pyvista/assets/89109579/8c7ddbc4-9eca-4daa-92b0-09af5593d56b)

But this error has no impact on any actual code execution. 

Similarly, if `mypy` is analyzing the code and cannot find the import, then the default assumption will be that the output type for these imports will be `Any`. I suppose this could be an issue if a downstream package had a very strict `mypy` config that disallowed missing imports or `Any` types, but if this was the case then there would also be many other errors generated anyway because most of the `pyvista` code base does not have annotations, and so already defaults to `Any`.

At most, if this actually does pose a problem for non-GL libraries somehow, I think adding a `type: ignore` for these lines somewhere would likely resolve it. Alternatively, would we could encase the imports in a `try-except` block, similar to the screenshot above for the `scipy, which I think may have a similar effect (I think `mypy` cannot fully resolve exceptions statically, and will assume variables have `Any` type).